lib/ejson/rails/railtie.rb

```diff
@@ -6,12 +6,17 @@ module Rails
     private_constant :Rails
 
     class Railtie < Rails::Railtie
+      singleton_class.attr_accessor(:set_secrets)
+      @set_secrets = true
+
       config.before_configuration do
-        json_files.each do |file|
-          next unless valid?(file)
+        json_file = json_files.detect { |file| valid?(file) }
+        next unless json_file
 
-          secrets = JSON.parse(file.read, symbolize_names: true)
-          break Rails.application.secrets.deep_merge!(secrets)
+        secrets = JSON.parse(json_file.read, symbolize_names: true)
+        Rails.application.secrets.deep_merge!(secrets) if set_secrets
+        Rails.application.credentials.config.deep_merge!(secrets) do |key|
+          raise "A credential already exists with the same name: #{key}"
```

Comment:
Would we want to raise a more specific error? `RuntimeError "A credential already exists with the same name: foo"` is probably enough for developers, though it could also be like `ExistingCredentialError`